msal_extensions/token_cache.py

```diff
@@ -1,156 +1,113 @@
-            self._last_sync = os.path.getmtime(self._cache_location)
+            self._persistence.save(self.serialize())
+            self._last_sync = time.time()
 
     def find(self, credential_type, **kwargs):  # pylint: disable=arguments-differ
         with CrossPlatLock(self._lock_location):
-            if self._needs_refresh():
-                try:
-                    self.deserialize(self._read())
-                except IOError as exp:
-                    if exp.errno != errno.ENOENT:
-                        raise
-                self._last_sync = time.time()
-            return super(FileTokenCache, self).find(credential_type, **kwargs)
-
-
-class UnencryptedTokenCache(FileTokenCache):
-    """An unprotected token cache to default to when no-platform specific option is available."""
-    def __init__(self, cache_location, **kwargs):
-        warnings.warn("You are using an unprotected token cache, "
-                      "because an encrypted option is not available for {}".format(sys.platform),
-                      RuntimeWarning)
-        super(UnencryptedTokenCache, self).__init__(cache_location, **kwargs)
-
+            self._reload_if_necessary()
+            return super(PersistedTokenCache, self).find(credential_type, **kwargs)
 
-class WindowsTokenCache(FileTokenCache):
-    """A SerializableTokenCache implementation which uses Win32 encryption APIs to protect your
-    tokens.
-    """
-    def __init__(self, cache_location, entropy='', **kwargs):
-        super(WindowsTokenCache, self).__init__(cache_location, **kwargs)
-        self._dp_agent = WindowsDataProtectionAgent(entropy=entropy)
-
-    def _write(self, contents):
-        with open(self._cache_location, 'wb') as handle:
-            handle.write(self._dp_agent.protect(contents))
-
-    def _read(self):
-        with open(self._cache_location, 'rb') as handle:
-            cipher_text = handle.read()
-        return self._dp_agent.unprotect(cipher_text)
 
+def build_token_cache(  # pylint: disable=too-many-arguments
+        location,
+        entropy='',  # For Windows
+        service_name="", account_name="",  # For OSX
+        schema_name="", attributes=None, fallback_to_plaintext=False,  # For Linux
+        **kwargs):  # Extra parameters will be relayed to LibsecretPersistence
+    """Build a token cache which uses native encryption mechanism on local platform.
 
-class OSXTokenCache(FileTokenCache):
-    """A SerializableTokenCache implementation which uses native Keychain libraries to protect your
-    tokens.
+    It will use DPAPI when running on Windows, Keychain when running on OSX,
+    libsecret when running on Linux and optionally fallback to plain texxt file.
     """
-
+    persistence = build_persistence(
+        location,
+        entropy=entropy,
+        service_name=service_name,
+        account_name=account_name,
+        schema_name=schema_name,
+        attributes=attributes or {"msal_token_cache": "1.0"},
```

Comment:
Another default value not defined in .NET or Java. I'm not able to verify this though. Do you know where this value comes from?

Reply:
Again, this is just some placeholder that I deliberately chose to be different. Perhaps we should even choose to NOT provide a default at all, and raise ValueError when app developer did not provide a specific value?

Reply:
I couldn't find this value in either the Java or .NET repo. Though in my Java testing I'm able to get Keyring work without providing this attribute.